src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -249,84 +249,91 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
+                        true, selectedValues, isDetailScreen);
+                if (!gotNextScreen) {
+                    notificationMessage = new NotificationMessage(
+                            "Overflowed selections with selection " + selection + " at index " + i,
+                            true,
+                            NotificationMessage.Tag.selection);
+                    break;
+                }
+                String nextInput = i == selections.length ? NO_SELECTION : selections[i];
+                Screen nextScreen;
+                try {
+                    nextScreen = autoAdvanceSession(menuSession, selection, nextInput, queryData,
+                            needsFullEntityScreen, inputValidated, forceManualAction, isDetailScreen);
+                } catch (CommCareSessionException e) {
+                    notificationMessage = new NotificationMessage(e.getMessage(), true,
+                            NotificationMessage.Tag.query);
+                    break;
+                }
+
+                if (nextScreen == null && menuSession.getSessionWrapper().getForm() == null) {
+                    // we've reached the end of this navigation path and no form in sight
+                    // this usually means a RemoteRequestEntry was involved
+                    if (nextInput != NO_SELECTION) {
+                        // still more nav to do so rebuild the session and continue
+                        executeAndRebuildSession(menuSession);
+                    } else {
+                        // no more nav, we're done
+                        BaseResponseBean postSyncResponse = resolveFormGetNext(menuSession);
+                        if (postSyncResponse == null) {
+                            // Return use to the app root
+                            postSyncResponse = new BaseResponseBean(null,
+                                    new NotificationMessage("Redirecting after sync", false,
+                                            NotificationMessage.Tag.sync),
+                                    true);
+                        }
+                        return postSyncResponse;
+                    }
+                } else {
+                    if (!selection.contentEquals(MultiSelectEntityScreen.USE_SELECTED_VALUES)) {
+                        menuSession.addSelection(selection);
+                    }
+                }
+            }
+
+            nextResponse = getNextMenu(
                     menuSession,
-                    null,
+                    detailSelection,
                     offset,
                     searchText,
                     sortIndex,
                     queryData,
                     casesPerPage,
                     smartLinkTemplate
             );
+            restoreFactory.cacheSessionSelections(menuSession.getSelections());
+        } catch (CommCareSessionException ccse) {
```

Comment:
Diff on this is not great, this catch is the main change here.

Reply:
how does this end up in Sentry?

Reply:
all the notifications tagged as errors gets logged through [this method](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/application/MenuController.java#L149).